wavefront_sdk/client.py

```diff
@@ -75,18 +75,18 @@ def __init__(self, server, token, max_queue_size=50000, batch_size=10000,
         self._tracing_spans_buffer = queue_impl(max_queue_size)
         self._spans_log_buffer = queue_impl(max_queue_size)
         self._events_buffer = queue_impl(max_queue_size)
-        self._headers = {'Content-Type': 'application/octet-stream',
-                         'Content-Encoding': 'gzip'}
-        self._event_headers = {'Content-Type': 'application/json',
-                               'Content-Encoding': 'gzip'}
+        self._headers = {'Content-Type': 'application/octet-stream'}
+        self._event_headers = {'Content-Type': 'application/json'}
         self._closed = False
         self._schedule_lock = threading.RLock()
         self._timer = None
         self._schedule_timer()
+        self._session = requests.Session()
```

Comment:
what's the lifetime of Client?  I assume it's for the entire runtime of a process?
If so, does having a single session for the entire lifetime work? 
https://stackoverflow.com/questions/59162353/python-requests-keep-connection-alive-for-indefinite-time and https://stackoverflow.com/questions/68404313/does-requests-session-expires-if-remained-idle
seems to suggest that the session stops working after the server closes the keep alive.  Did you try this out?

If the session does expire, then perhaps you can just create a Session every `_internal_flush`, use it as a context manager in that function and just pass it down to `_batch_report` and `_report`.   I guess a session every 5 seconds isn't bad.

The other thing you could do if sessions do expire is just `try...except` it and renew the session in the _except_ loop

Reply:
I read https://requests.readthedocs.io/en/latest/user/advanced/#session-objects that Session objects are supposed to take care of bringing up a new connection underneath

So I ran a slightly modified version of https://github.com/wavefrontHQ/wavefront-sdk-python/blob/master/example.py overnight, with 
```
     wavefront_client.send_metric(
        'python.proxy.new york.power.usage',
        time.time() % 3600, None, 'austinov_mac',
        {'wf_client_id': str(id(wavefront_client)),
         'session_id': str(id(wavefront_client._session))})
```
         
And I've got quite interesting results https://qa4.wavefront.com/u/VS3wr4ZSDJ?t=collector
(delta counter https://qa4.wavefront.com/u/584jl40QZz?t=collector)    
which show the connection seemed to be kept alive overnight, but I guess was affected by sleep policy of my laptop

Reply:
So the process on my laptop resumed sending points after sleeping in my backpack during my commute https://qa4.wavefront.com/u/YNQlqRlTg6?t=collector so it looks like `requests.Session()` is taking a good care of reconnecting